portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,204 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+/*
+	*Dynamically fetch content from github and render it to html
+
+    *Description of functions below:
+        *getData() asyncronously fetched data from github and calls functions to parse, filter and render component\
+        *SetHeader(repo) fetches and returns certain atributres from repo object
+        *SetRepos(repo) fetches and returns certain atributres from repo object
+        *injectHeader(header) dynamically renders the header to the body
+        *injectRepo(repo) creates a new repository elements and injects it into the body
+
+*/
+
+async function getData (){
+    await fetch('https://api.github.com/users/MohamedShatry/repos')
+            .then(data => data.json())
+            .then(res => {
+                let varHeader = SetHeader(res[0].owner);
```

Comment:
This might crash if len(res) = 0

Reply:
I'm thinking of ways to catch that. I'm already catching any error that the fetch call would return, but would adding this fix the problem? 

if(res.length === 0){
                    injectError();
                    next();
                }

Reply:
Resolved